src/inference/src/compilation_context.cpp

```diff
@@ -65,69 +139,58 @@ std::string NetworkCompilationContext::calculateFileInfo(const std::string& file
+                                                    const ov::AnyMap& compileOptions) {
+    OV_ITT_SCOPE(FIRST_INFERENCE, ov::itt::domains::IE_RT, "NetworkCompilationContext::compute_hash - Model");
+
+    return std::to_string(compute_model_hash(model, compileOptions));
+}
+
+std::string NetworkCompilationContext::compute_hash(const InferenceEngine::CNNNetwork& network,
+                                                    const ov::AnyMap& compileOptions) {
+    OV_ITT_SCOPE(FIRST_INFERENCE, ov::itt::domains::IE_RT, "NetworkCompilationContext::compute_hash - CNN");
 
-    IE_ASSERT(network.getFunction());
+    OPENVINO_ASSERT(network.getFunction());
 
     uint64_t seed = 0;
     // 1. Calculate hash on function
-    ov::pass::Manager m;
-    m.register_pass<ngraph::pass::FixRtInfo>();
-    m.register_pass<ov::pass::Hash>(seed);
-    m.run_passes(network.getFunction());
-
     // 2. Compute hash on serialized data and options
-    for (const auto& kvp : compileOptions) {
-        seed = hash_combine(seed, kvp.first + kvp.second);
-    }
-
     // 3. Add runtime information which may not be serialized
-    for (const auto& op : network.getFunction()->get_ordered_ops()) {
-        const auto& rt = op->get_rt_info();
-        for (const auto& rtMapData : rt) {
-            seed = hash_combine(seed, rtMapData.first);
-            std::stringstream strm;
-            rtMapData.second.print(strm);
-            seed = hash_combine(seed, strm.str());
-        }
-    }
+    seed = compute_model_hash(network.getFunction(), compileOptions);
 
     // 4. Add inputs info
     for (const auto& input : network.getInputsInfo()) {
-        InputInfo::Ptr info = input.second;
+        InferenceEngine::InputInfo::Ptr info = input.second;
         seed = hash_combine(seed, as_int32_t(info->getPrecision()));
         seed = hash_combine(seed, as_int32_t(info->getLayout()));
 
         const InferenceEngine::PreProcessInfo& preproc = info->getPreProcess();
         seed = hash_combine(seed, as_int32_t(preproc.getMeanVariant()));
 
-        if (preproc.getMeanVariant() == MeanVariant::MEAN_VALUE) {
+        if (preproc.getMeanVariant() == InferenceEngine::MeanVariant::MEAN_VALUE) {
             seed = hash_combine(seed, preproc.getNumberOfChannels());
             for (size_t c = 0; c < preproc.getNumberOfChannels(); ++c) {
-                const PreProcessChannel::Ptr& channelInfo = preproc[c];
+                const InferenceEngine::PreProcessChannel::Ptr& channelInfo = preproc[c];
                 seed = hash_combine(seed, channelInfo->stdScale);
                 seed = hash_combine(seed, channelInfo->meanValue);
             }
-        } else if (preproc.getMeanVariant() == MeanVariant::MEAN_IMAGE) {
+        } else if (preproc.getMeanVariant() == InferenceEngine::MeanVariant::MEAN_IMAGE) {
             // TODO: think if we need to compute hash for mean image if it exists
         }
```

Comment:
duplicated code between compute_hash with CNNNetwork and ov::Model. Should we extract hash compute on `InferenceEngine::InputInfo` to a separate function and reuse in both places?

Reply:
Fixed